File: Python3.x/13-Roman_to_Integer.py
# Runtime: 56 ms, faster than 95.24% of Python3 online submissions for Roman to Integer.
# Memory Usage: 13.3 MB, less than 61.58% of Python3 online submissions for Roman to Integer.
class Solution:
    def romanToInt(self, s: str) -> int:
        if len(s) == 0: return 0
        dictory = {'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000}
        s, res, i = list(s), 0, 0
        for i, val in enumerate(s):
            s[i] = dictory[s[i]]
        len_s = len(s)
        for i, _ in enumerate(s[:-1]):
                res = res - s[i] if s[i] < s[i + 1] else res + s[i]
        res += s[-1]
        return res

File: Algorithms/C++/13-Roman_to_Integer.cpp
// Runtime: 28 ms, faster than 13.11% of C++ online submissions for Roman to Integer.
// Memory Usage: 10.6 MB, less than 22.55% of C++ online submissions for Roman to Integer.
class Solution {
public:
    int romanToInt(string s) {
        map<char, int> sym_val = {
            {'I', 1},
            {'V', 5},
            {'X', 10},
            {'L', 50},
            {'C', 100},
            {'D', 500},
            {'M', 1000},
        };
        int result = sym_val[s[s.size() - 1]];
        for (int i = s.size() - 2; i >= 0; --i) {
            if (sym_val[s[i]] < sym_val[s[i + 1]])
                result -= sym_val[s[i]];
            else
                result += sym_val[s[i]];
        }
        return result;
    }
};
